### ISSUE TITLE
Filter: buttons missing attribute to communicate they have popup (4.1.2)

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux1#869]
### Expected behavior

Buttons that open the filter modal dialog have aria-haspopup="dialog". This is both filter button and the filter pill buttons. https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup.
### Actual behavior

aria-haspopup attribute missing for buttons.
Pill buttons also have label text = "close". "Remove" would better text equivalent for this behavior in this close.
